dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/xds/PilotExchanger.java

```diff
@@ -0,0 +1,184 @@
+
+    private static PilotExchanger GLOBAL_PILOT_EXCHANGER = null;
+
+    private final ApplicationModel applicationModel;
+
+    private static final Map<String, XdsVirtualHost> xdsVirtualHostMap = new ConcurrentHashMap<>();
+
+    private static final Map<String, XdsCluster> xdsClusterMap = new ConcurrentHashMap<>();
+
+    private final Map<String, Set<XdsDirectory>> rdsListeners = new HashMap<>();
+
+    private final Map<String, Set<XdsDirectory>> cdsListeners = new HashMap<>();
+
+    protected PilotExchanger(URL url) {
+        // xdsChannel = new XdsChannel(url);
+        int pollingTimeout = url.getParameter("pollingTimeout", 10);
+        this.applicationModel = url.getOrDefaultApplicationModel();
+        adsObserver = new AdsObserver(url, NodeBuilder.build());
+
+        // rds 资源回调函数，将 RdsProtocol 的资源存放起来
+        Consumer<List<XdsRouteConfiguration>> rdsCallback = (xdsRouteConfigurations) -> {
+            System.out.println(xdsRouteConfigurations);
+            xdsRouteConfigurations.forEach(xdsRouteConfiguration -> {
+                xdsRouteConfiguration.getVirtualHosts().forEach((serviceName, xdsVirtualHost) -> {
+                    this.xdsVirtualHostMap.put(serviceName, xdsVirtualHost);
+                    // 回调更新
+                    if (rdsListeners.containsKey(serviceName)) {
+                        for (XdsDirectory listener : rdsListeners.get(serviceName)) {
+                            listener.onRdsChange(serviceName, xdsVirtualHost);
+                        }
+                    }
+                });
+            });
+        };
+
+        // eds 资源回调函数
+        Consumer<List<XdsCluster>> edsCallback = (xdsClusters) -> {
+            System.out.println(xdsClusters);
+            xdsClusters.forEach(xdsCluster -> {
+                this.xdsClusterMap.put(xdsCluster.getName(), xdsCluster);
+                if (cdsListeners.containsKey(xdsCluster.getName())) {
+                    for (XdsDirectory listener : cdsListeners.get(xdsCluster.getName())) {
+                        listener.onCdsChange(xdsCluster.getName(), xdsCluster);
+                    }
+                }
+            });
+        };
+
+        this.ldsProtocol = new LdsProtocol(adsObserver, NodeBuilder.build(), pollingTimeout);
+        this.rdsProtocol = new RdsProtocol(adsObserver, NodeBuilder.build(), pollingTimeout, rdsCallback);
+        this.edsProtocol = new EdsProtocol(adsObserver, NodeBuilder.build(), pollingTimeout, edsCallback);
+        this.cdsProtocol = new CdsProtocol(adsObserver, NodeBuilder.build(), pollingTimeout);
+
+        // lds 回调函数，在回调函数中监听所有的 rds 资源
+        Consumer<Set<String>> ldsCallback = (routes) -> {
+            rdsProtocol.getResource(routes);
+        };
+
+        ldsProtocol.setUpdateCallback(ldsCallback);
+
+        ldsProtocol.getListeners();
+
+        // cds 回调函数，在回调函数中监听所有的 eds 资源
+        Consumer<Set<String>> cdsCallback = (clusters) -> {
+            edsProtocol.getResource(clusters);
```

Comment:
also don't have receiver

Reply:
The function is used to listen to Cds resources. The name of the function has been changed.